### PR TITLE
Develop/imx95 support

### PR DESCRIPTION
**Pull request by Zhiqiang-Hou**
Add HAL support for imx95

### CODE REVIEW

File: dts/nxp/nxp_imx/mimx9596avzxn-pinctrl.dtsi
Comment (JiafeiPan) [inline dts/nxp/nxp_imx/mimx9596avzxn-pinctrl.dtsi:6]: this is a temporary file and not generated from MIMX8ML8DVNLZ ?

File: mcux/CMakeLists.txt
Comment (JiafeiPan) [inline mcux/CMakeLists.txt:46]: reuse CONFIG_HAS_IMX_SM?
zephyr_compile_definitions_ifdef(CONFIG_HAS_IMX_SM FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL=1)

**Review by JiafeiPan — CHANGES_REQUESTED** [review]
Most of the update has been merged into mcux-sdk: https://github.com/nxp-mcuxpresso/mcux-sdk/commits/main.

but let's refine some hal cmake, thanks.

File: mcux/CMakeLists.txt
Comment (Zhiqiang-Hou) [reply mcux/CMakeLists.txt:46]: I prefer the CONFIG_CLOCK_CONTROL_IMX_SCMI, currently the implementation leverages the imx-sm SCMI transport/protocol layers, but finally, as plan, we will implement SCMI framwork on zephyr and drop the imx-sm code.

File: mcux/CMakeLists.txt
Comment (Zhiqiang-Hou) [reply mcux/CMakeLists.txt:87]: As Laurentiu has worked out a general SCMI framework support for Zephyr (https://github.com/zephyrproject-rtos/zephyr/pull/75102), we will move to use it and drop the imx-sm code.

**Comment by dbaluta** [comment]
@Zhiqiang-Hou as a general comment please try to other more description in the commit message to help reviewers. Please try to focus on reasoning on why every commit is needed and also explain the inner details where is it needed.

File: mcux/mcux-sdk/tools/cmake_toolchain_files/armgcc_aarch64.cmake
Comment (dbaluta) [inline mcux/mcux-sdk/tools/cmake_toolchain_files/armgcc_aarch64.cmake:48]: Care to explain more in the commit message why this commit is needed. What is the actual update? I could see that you add floating point option but it is not clear from the commit message.

File: mcux/mcux-sdk/devices/MIMX9596/drivers/ChangeLogKSDK.txt
Comment (dbaluta) [inline mcux/mcux-sdk/devices/MIMX9596/drivers/ChangeLogKSDK.txt:2]: What is mcu-soc in the commit message. If this is another repo unrelated to zephyr can you please add a link to the repo?

File: mcux/mcux-sdk/devices/MIMX9596/MIMX9596_ca55_features.h
Comment (dbaluta) [inline mcux/mcux-sdk/devices/MIMX9596/MIMX9596_ca55_features.h]: Why? Can you please explain in the commit message.

**Comment by yangbolu1991** [comment]
@JiafeiPan @dbaluta 
As we will be based on ARM SCMI https://github.com/zephyrproject-rtos/zephyr/pull/75102, 
the i.MX95 supoort in hal becomes much easy.

I updated the pull request with only a few of files changes.
Please help to review.

Thank you very much.

**Comment by yangbolu1991** [comment]
Any comments?
Thanks.

**Review by dbaluta — APPROVED** [review]

**Comment by dbaluta** [comment]
Looks good to me AFAICT!

**Review by JiafeiPan — APPROVED** [review]
Fine for me, thanks.

**Comment by JiafeiPan** [comment]
Please rebase this PR as sdk 1.6 update is merged.

**Comment by yangbolu1991** [comment]
Updated to v3. Changes,
- Dropped mu1 driver sync patch since code base already had the update.

Thanks.

**Review by LaurentiuM1234 — APPROVED** [review]

**Review by dbaluta — APPROVED** [review]

**Review by dleach02 — CHANGES_REQUESTED** [review]
Folks, the safe process here is to make sure the Zephyr side PR is ready and clean to merge before we merge the HAL side. We have too much activity with HAL/Zephyr PRs to risk merging this then having a later PR merge and update Zephyr's manifest to a sha that contains these changes. I get that this appears fairly isolated but that is the general safe process. 

I'm going to put a block on this until you have a Zephyr side PR ready to merge to avoid accidental merging.

**Comment by LaurentiuM1234** [comment]
@yangbolu1991 @Zhiqiang-Hou can you pls rebase this? I suspect some twister failures in https://github.com/zephyrproject-rtos/zephyr/pull/74920 are caused by this.

**Comment by Zhiqiang-Hou** [comment]
Sure, thanks for the suggestion!

**Comment by Zhiqiang-Hou** [comment]
v4:
- Rebased.

**Comment by yangbolu1991** [comment]
Updated to v5. Changes,
- Rebased.

Thanks.

**Review by dbaluta — APPROVED** [review]

**Comment by JiafeiPan** [comment]
@dleach02,  it seems Zephyr PR has already been settled down,  could we merge this hal PR firstly? thanks.

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**
